packages/legend-graph/src/models/protocols/pure/v1/transformation/pureProtocol/serializationHelpers/V1_RuntimeSerializationHelper.ts

```diff
@@ -71,40 +74,43 @@ export const V1_legacyRuntimeModelSchema = createModelSchema(V1_LegacyRuntime, {
   ),
 });
 
-export const V1_identifiedConnectionModelSchema = createModelSchema(
-  V1_IdentifiedConnection,
-  {
+export const V1_identifiedConnectionModelSchema = (
```

Comment:
please don't do this. The constant name is `V1_identifiedConnectionModelSchema` which is not supposed to be a schema, nor is it a schema creator, what you are doing here is essentially similar to what you did with runtime where you should either:
- create a pair of `deserialize`, `serialize` methods for identified connection
- OR, do what you did with `setupEngineRuntime`

Reply:
I'd prefer the first (and this means propagating the `plugins` from `V1_setupEngineRuntimeSerialization` just because this is really minor stuff and not getting exported to be used outside the context of runtime

Reply:
I don't get what you mean by: ' V1_identifiedConnectionModelSchema which is not supposed to be a schema, nor is it a schema creator'

Reply:
export const V1_identifiedConnectionSerialization = (
  plugins: PureProtocolProcessorPlugin[],
): ModelSchema<V1_IdentifiedConnection> =>
  createModelSchema(V1_IdentifiedConnection, {
    connection: custom(
      (val) => V1_serializeConnectionValue(val, true, plugins),
      (val) => V1_deserializeConnectionValue(val, true, plugins),
    ),
    id: primitive(),
  });

export const V1_storeConnectionSerialization = (
  plugins: PureProtocolProcessorPlugin[],
): ModelSchema<V1_StoreConnections> =>
  createModelSchema(V1_StoreConnections, {
    store: usingModelSchema(V1_packageableElementPointerDeserrializerSchema),
    storeConnections: list(
      custom(
        val => serialize(V1_identifiedConnectionSerialization(plugins), val),
        val => deserialize(V1_identifiedConnectionSerialization(plugins), val),
      )
    ),
  });

export const V1_setupEngineRuntimeSerialization = (
  plugins: PureProtocolProcessorPlugin[],
):void => {
  createModelSchema(V1_EngineRuntime, {
    _type: usingConstantValueSchema(V1_RuntimeType.ENGINE_RUNTIME),
    connections: custom(
      val => serialize(V1_storeConnectionSerialization(plugins), val),
      val => deserialize(V1_storeConnectionSerialization(plugins), val),
    ),
    mappings: list(
      usingModelSchema(V1_packageableElementPointerDeserrializerSchema),
    ),
  });
};

Reply:
no, I mean something like

```ts
export const V1_setupEngineRuntimeModelSchema = (plugins): void => {
  createModelSchema(V1_IdentifiedConnection, {
    connection: custom(
      (val) => V1_serializeConnectionValue(val, true),
      (val) => V1_deserializeConnectionValue(val, true),
    ),
    id: primitive(),
  });
  createModelSchema(V1_StoreConnections, {
    store: usingModelSchema(V1_packageableElementPointerDeserrializerSchema),
    storeConnections: list(object(V1_IdentifiedConnection)),
  });
  createModelSchema(V1_EngineRuntime, {
    _type: usingConstantValueSchema(V1_RuntimeType.ENGINE_RUNTIME),
    connections: list(object(V1_StoreConnections)),
    mappings: list(
      usingModelSchema(V1_packageableElementPointerDeserrializerSchema),
    ),
  });
};
```